packages/shared/src/components/tooltips/InteractivePopup.tsx

```diff
@@ -41,30 +45,83 @@ const positionClass: Record<InteractivePopupPosition, string> = {
   leftStart: classNames(leftClass, startClass),
   leftCenter: classNames(leftClass, centerClassY),
   leftEnd: classNames(leftClass, endClass),
+  profileMenu: classNames(profileMenuRightClass, startClass),
   screen: 'inset-0 w-screen h-screen',
-  mainFeedLeftEnd: classNames(mainFeedLeftClass, mainFeedEndClass),
 };
 
+const leftPositions = [
+  InteractivePopupPosition.LeftEnd,
+  InteractivePopupPosition.LeftCenter,
+  InteractivePopupPosition.LeftEnd,
+];
+
 function InteractivePopup({
   children,
   className,
   position = InteractivePopupPosition.Center,
+  closeOutsideClick,
+  onClose,
   ...props
 }: InteractivePopupProps): ReactElement {
-  const classes = positionClass[position];
+  const container = useRef<HTMLDivElement>();
+  const onCloseRef = useRef(onClose);
+  const { sidebarRendered } = useSidebarRendered();
+  const { sidebarExpanded } = useSettingsContext();
+  const finalPosition = sidebarRendered
+    ? position
+    : InteractivePopupPosition.Center;
+  const classes = positionClass[finalPosition];
+
+  useEffect(() => {
+    if (!closeOutsideClick || !onCloseRef?.current || !container?.current) {
+      return null;
+    }
+
+    const onClickAnywhere = (e: MouseEvent) => {
+      if (!container.current.contains(e.target as Node)) {
+        onCloseRef.current(e);
+      }
+    };
+
+    globalThis?.document.addEventListener('click', onClickAnywhere);
```

Comment:
Cant we move this to useWindowEvents now?

Reply:
Currently, we should only enable it when the props is right, but we currently don't support it in that hook. Meaning, the listener will be added even though it is not needed.

Reply:
Hmm should we modify it perhaps?
I really would like to reuse the hook we have to keep complex edge case friendly hooks uniform

Reply:
I want to, but it will involve a lot of refactors on existing implementations with the type to make work with `click` as I mentioned in our discussion. That's why I was suggesting to open a ticket dedicated to it.

Reply:
I send you a DM on slack with an example I created, that works but love to hear what you think.